### ISSUE TITLE
SDL3, macOS: SDL_HINT_RENDER_DRIVER fallback does not work

### ISSUE THREAD

**Issue by andreasgrabher** [libsdl-org/SDL#11077]
I tried setting SDL_HINT_RENDER_DRIVER to "direct3d" on macOS just to see if it will fall back to the default renderer. Unfortunately this seems to not work as described in the documentation:

> If the application doesn't pick a specific renderer to use, this variable
> specifies the name of the preferred renderer. If the preferred renderer
> can't be initialized, the normal default renderer is used.

I get this error when calling `SDL_CreateRenderer()`:

> Couldn't find matching render driver!

**Comment by slouken** [comment]
Oh, the documentation is wrong here. I'll fix that. It's intentional that it returns NULL if the preferred renderer isn't available, so that the application can try several, in order of preference.

**Comment by andreasgrabher** [comment]
I am not sure this makes sense. If I want to explicitly create a metal renderer I just call 
```
SDL_CreateRenderer(sdlRenderer, "metal");
```

Why should someone call 
```
SDL_SetHint(SDL_HINT_RENDER_DRIVER, "metal");
SDL_CreateRenderer(sdlRenderer, NULL);
```
instead? I think if the hint works this way it is pretty much useless.

**Comment by slouken** [comment]
Because you can do this:
```c
renderer = SDL_CreateRenderer("gpu");
if (!renderer) {
    renderer = SDL_CreateRenderer("opengles2");
}
if (!renderer) {
    renderer = "SDL_CreateRenderer("direct3d");
}
```

**Comment by andreasgrabher** [comment]
Exactly, so why would I need that hint?

**Comment by madebr** [comment]
Isn't a hint useful for a user to try another renderer backend (through an environment variable), and the `SDL_CreateRenderer` a way to force a backend by the developer?

**Comment by andreasgrabher** [comment]
I think this hint would make sense it someone has one single preferred renderer but does not want things to fail if that one is not available.

I think if someone wants to make his own order of preference with multiple renderers this would be appropriate:

```
static const char* renderers[] = {
	"direct3d",
	"direct3d11",
	"direct3d12",
	"metal",
	"opengl",
	"opengles2",
	"opengles",
	"vulkan",
	"software"
};

SDL_Renderer* CreateRenderer(SDL_Window* sdlWindow)
{
	int i;
	SDL_Renderer* sdlRenderer;
	for (i = 0; i < SDL_arraysize(renderers); i++) {
		fprintf(stderr, "Trying to create %s renderer\n", renderers[i]);
		sdlRenderer = SDL_CreateRenderer(sdlWindow, renderers[i]);
		if (sdlRenderer) {
			fprintf(stderr, "Successfully created %s renderer\n", SDL_GetRendererName(sdlRenderer));
			return sdlRenderer;
		}
	}
	return NULL;
}
```

**Comment by slouken** [comment]
Yep, that's what we do for video drivers, etc. I'll take a look.

**Comment by icculus** [comment]
(Just making a note that what we need to do here is make sure SDL_HINT_RENDER_DRIVER takes a comma-separated list, to match what audio and video subsystems already do.)

**Comment by andreasgrabher** [comment]
I know that 3.2.0 should be released really soon, but I think this shoud be fixed before release. People might use this hint to set a specific renderer instead of using SDL_CreateRenderer() for that. Changing the hint to be treated as prefered renderer could then break someone's code.

**Comment by icculus** [comment]
I snuck this in for 3.2.0.

The documentation in the original comment was updated at some point to reflect reality, but this now accepts a comma-separated list (both in SDL_CreateRenderer's `name` parameter and SDL_HINT_RENDER_DRIVER), matching what the video/audio/whatever subsystems already do.

**Comment by andreasgrabher** [comment]
Fantastic! Thank you very much!
